File: task/task/src/main/java/com/Task_Management/task/models/Task.java
package com.Task_Management.task.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.FutureOrPresent;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String title;

    private String description;

    @FutureOrPresent
    private LocalDate dueDate;

    private boolean completed = false;

    @ManyToOne
    private User assignedUser;

    @ManyToOne
    private User createdBy;

    private String priority; // HIGH, MEDIUM, LOW

    @PrePersist
    private void setDefaults() {
        if (priority == null) {
            priority = "MEDIUM";
        }
    }
}

File: task_management/task_management/lib/models/task.dart

import 'package:task_management/models/user.dart';

class Task {
  final int id;
  final String title;
  final String description;
  final DateTime dueDate;
  final String priority;
  final bool isCompleted;
  final User? assignedTo;

  Task({
    required this.id,
    required this.title,
    required this.description,
    required this.dueDate,
    required this.priority,
    required this.isCompleted,
    this.assignedTo,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      dueDate: DateTime.parse(json['dueDate']),
      priority: json['priority'],
      isCompleted: json['isCompleted'],
      assignedTo: json['assignedTo'] != null ? User.fromJson(json['assignedTo']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'dueDate': dueDate.toIso8601String(),
      'priority': priority,
      'isCompleted': isCompleted,
      'assignedTo': assignedTo?.toJson(),
    };
  }
}
